src/tpm2.c

```diff
@@ -191,21 +288,38 @@ static TPM_RC TPM2_SendCommandAuth(TPM2_CTX* ctx, TPM2_Packet* packet,
         /* Do encryption of the first parameter in the command request */
         sessionParamEnc = TPM2_ParamEnc_FindDecryptSession(ctx);
         if (sessionParamEnc < MAX_SESSION_NUM) {
+        #ifndef WOLFTPM2_NO_WOLFCRYPT
+            TPMS_AUTH_COMMAND authHmac[MAX_SESSION_NUM];
+        #endif
             rc = TPM2_ParamEnc_CmdRequest(&ctx->authCmd[sessionParamEnc],
                                           &encryptedParameter,
                                           param, requestParamSz);
             if (rc != TPM_RC_SUCCESS) {
 #ifdef DEBUG_WOLFTPM
                 printf("Request parameter encryption failed\n");
 #endif
-                return TPM_RC_FAILURE;
+                return rc;
             }
+
             /* packet->pos already points at the first command parameter */
             TPM2_Packet_AppendBytes(packet, encryptedParameter.buffer,
-                                    encryptedParameter.size);
+                                            encryptedParameter.size);
+
+        #ifndef WOLFTPM2_NO_WOLFCRYPT
```

Comment:
Could parameter encryption work without this hmac field calculation?
My read: if this hmac filed is indeed necessary for policy or hmac session, then it is necessary for parameter encryption. Because parameter encryption requires at policy session or hmac. I would recommend not just adding build option for param enc, but adding a comment here about this relation.